### PR TITLE
Added a move to the MediaHandler and a null test

### PR DESCRIPTION
**Pull request by stazio**

### CODE REVIEW

File: src/impl/track.cpp [hunk 1]
```diff
@@ -182,10 +182,11 @@ bool Track::transportSend([[maybe_unused]] message_ptr message) {
 void Track::setMediaHandler(shared_ptr<MediaHandler> handler) {
 	{
 		std::unique_lock lock(mMutex);
```
Comment (stazio) [inline src/impl/track.cpp:184]: Should this function `mRtcpHandeler->onOutgoing(nullptr)` before setting the new callback?

Comment (paullouisageneau) [reply src/impl/track.cpp:184]: Yes, I think it could be good to ensure they is only one handler outputting something.

**Review by paullouisageneau — APPROVED** [review]

File: src/impl/track.cpp [hunk 2]
```diff
@@ -182,10 +182,11 @@ bool Track::transportSend([[maybe_unused]] message_ptr message) {
...
-		mMediaHandler = handler;
+		mMediaHandler = std::move(handler);
 	}
 
-	handler->onOutgoing(std::bind(&Track::transportSend, this, std::placeholders::_1));
+	if (mMediaHandler)
+		mMediaHandler->onOutgoing(std::bind(&Track::transportSend, this, std::placeholders::_1));
```
Comment (paullouisageneau) [inline src/impl/track.cpp:189]: This operates on `mMediaHandler` outside the lock and therefore breaks synchonization. Fixed in https://github.com/paullouisageneau/libdatachannel/pull/369.

File: src/impl/track.cpp (hunk 1 above)
Comment (paullouisageneau) [reply src/impl/track.cpp:184]: Added in https://github.com/paullouisageneau/libdatachannel/pull/369.

**Pull request merged**